packages/api/models/rules.js

```diff
@@ -498,7 +513,23 @@ class Rule extends Manager {
     const subscriptionParams = {
       SubscriptionArn: item.rule.arn,
     };
-    return awsServices.sns().unsubscribe(subscriptionParams).promise();
+    return this.SnsClient.unsubscribe(subscriptionParams).promise();
+  }
+
+  async deleteOldEventSourceMappings(item) {
+    switch (item.rule.type) {
+    case 'kinesis':
+      await this.deleteKinesisEventSources(item);
+      break;
+    case 'sns': {
+      if (item.rule.arn) {
+        await this.deleteSnsTrigger(item);
+      }
+      break;
+    }
+    default:
+      break;
+    }
```

Comment:
What about scheduled rules? This function is fine but maybe you can refactor `delete` instead since a lot of the logic overlaps?

Reply:
So I was initially tempted to do that refactor, but I decided to hold off because I think the intents of `deleteOldEventSourceMappings` and `delete` are somewhat different. `deleteOldEventSourceMappings` is there to remove the old trigger mappings for SNS/Kinesis rules after they have been updated with new triggers. For scheduled rules, I don't think there is any cleanup necessary for when the rule is updated because it fully overwrites the Cloudwatch event:

https://github.com/nasa/cumulus/blob/master/packages/api/models/rules.js#L41-L57

In the case of `delete`, we do want to cleanup everything including Cloudwatch events since we're deleting the rule itself. 

Does that make sense? Would it be helpful to rename these functions for clarity?

Reply:
Ah, I didn't realize that the Cloudwatch events get overwritten so they don't need to get deleted but I see it now. In that case, I think keeping `deleteOldEventSourceMappings` is fine. In phase 3, I renamed `delete` to `deleteRuleResources` https://github.com/nasa/cumulus/blob/feature/rds-phase-3/packages/api/lib/rulesHelpers.js#L237